src/js/hooks/list-pages/productSupplier/useProductSupplierListTableData.jsx

```diff
@@ -1,25 +1,43 @@
-import { useState } from 'react';
+import _ from 'lodash';
 
 import { PRODUCT_SUPPLIER_API } from 'api/urls';
 import useTableData from 'hooks/list-pages/useTableData';
 
-const useProductSupplierListTableData = () => {
+const useProductSupplierListTableData = (filterParams) => {
   const errorMessageId = 'react.productSupplier.error.productSupplierList.label';
   const defaultErrorMessage = 'Unable to fetch product sources';
 
   const defaultSorting = {
     sort: 'dateCreated',
-    order: 'asc',
+    order: 'desc',
   };
 
-  // TODO: To be removed after adding filters
-  const [filterParams] = useState({ offset: 0, max: 10 });
-
-  const getParams = () => {};
+  const getParams = ({
+    offset,
+    state,
+    sortingParams,
+  }) => {
+    const {
+      product,
+      supplier,
+    } = filterParams;
+    return _.omitBy({
+      offset: `${offset}`,
+      max: `${state.pageSize}`,
+      ...sortingParams,
+      ...filterParams,
```

Comment:
probably it's based on my preferences, but I prefer spreading parameters at the end of the object (do not treat this as a change request)

Reply:
if I had done it this way, it would eventually override the `product` and the `supplier` properties.

Reply:
For my whole life, I was sure that this thirdObj should return "thirdObj". I was sure that the parameters from spreading were always overridden by those typed "directly" 🤡🤡
![image](https://github.com/openboxes/openboxes/assets/83239466/c3ac8e6a-f7c1-4795-930a-7f42b59c4741)